trieste/acquisition/function/greedy.py

```diff
@@ -16,17 +16,21 @@
 """
 from __future__ import annotations
 
-from typing import Callable, Optional, Union, cast
+from typing import Callable, Mapping, Optional, Union, cast
 
+import gpflow
 import tensorflow as tf
 import tensorflow_probability as tfp
 
 from ...data import Dataset
 from ...models import ProbabilisticModel
+from ...models.gpflow import GaussianProcessRegression
```

Comment:
this is where the new interface `FastUpdateModel` comes in, we can use it instead of `GaussianProcessRegression`